src/twisted/protocols/test/test_ftp

```diff
@@ -0,0 +1,33 @@
+import mock
+import twisted
+from twisted.trial import unittest
+from twisted.protocols.ftp import ASCIIConsumerWrapper
+
+class ASCIIConsumerWrapperTestCase(unittest.TestCase):
+    """ Test case for ASCIIConsumerWrapper, tests __init__ and write function ""
+    @mock.patch("twisted.protocols.ftp.os")
+    def test_init(self, os_mock):
+        """ Test non supported platform, and make sure that for '\r\n' we're not wrapping the write function.
+        cons_mock = mock.MagicMock()
```

Comment:
Don't use a mock for this &mdash; especially not a `MagicMock`, the footgunniest of mocks. I'd suggest defining a class right here in the test, but it looks like `twisted.protocols.ftp.FileConsumer` will do, like:

```py
buf = io.BytesIO()
wrapper = ASCIIConsumerWrapper(FileConsumer(buf))

wrapper.write(f"test{os.linesep}".encode())

self.assertEqual(b'\r\n', buf.getvalue())
```